src/lib/Quiz.jsx

```diff
@@ -26,28 +78,41 @@ const Quiz = function ({
   }, []);
 
   useEffect(() => {
+    // declaring a common variale for all operation
+    let newQuestions = quiz.questions;
+
+    // shuffling questions  if shuffle is true
     if (shuffle) {
-      setQuestions(shuffleQuestions(quiz.questions));
-    } else {
-      quiz.questions.length = nrOfQuestions;
-      setQuestions(quiz.questions);
+      newQuestions = shuffleQuestions(newQuestions);
     }
 
-    setQuestions(questions.map((question, index) => ({
+    // shuffling answers if shuffleAnswer is true
+    if (shuffleAnswer) {
+      newQuestions = shuffleAnswerSequence(newQuestions);
+    }
+
+    // setting number of question by defining array length(used min function for if )
```

Comment:
dont understand what it means by `(used min function for if )`

Reply:
at first i tried to do this
![image](https://github.com/wingkwong/react-quiz-component/assets/75330767/1435ca40-d0d2-4428-a79a-f228a1aef718)
then i realised you have already validate the "nrOfQuestions" variable above.
so i deleted the Math.min section and kept as it was, but forgot to delete the comment